test/lib/convert-forms/handle-db-doc-ref.spec.js

```diff
@@ -0,0 +1,59 @@
+const { expect } = require('chai');
+const { handleDbDocRefs } = require('../../../src/lib/convert-forms/handle-db-doc-ref');
+const { createXformDoc, createXformString, serializeToString, FORM_ID } = require('../../fn/convert-forms.utils');
+
+describe('Handle db-doc-ref attributes', () => {
+  it('replaces form references with references to the primary instance node name', () => {
+    const doc = createXformDoc({
+      primaryInstance: `<test db-doc-ref="/${FORM_ID}"/>`,
+    });
+
+    handleDbDocRefs(doc);
+
+    const expectedDoc = createXformString({
+      primaryInstance: `<test db-doc-ref="/data"/>`,
+    });
+    expect(serializeToString(doc)).xml.to.equal(expectedDoc);
+  });
+
+  it('replaces relative paths with absolute paths', () => {
+    const doc = createXformDoc({
+      primaryInstance: `
+        <test db-doc-ref="../group/other"/>
+        <group>
+          <other db-doc-ref="../../test"/>
+        </group>        
+      `,
+    });
+
+    handleDbDocRefs(doc);
+
+    const expectedDoc = createXformString({
+      primaryInstance: `
```

Comment:
aside: wow serious circular referencing here! is this even allowed in enketo forms?